File: Array/011-SingleNumber/SingleNumber.go
func singleNumber(nums []int) int {
    numMap := make(map[int]int)
    
    for i:=0; i<len(nums); i++ {
        _, exist := numMap[nums[i]]
        if exist {
            delete(numMap, nums[i])
        } else {
            numMap[nums[i]] = 1
        }
    }
    
    for key := range numMap {
        return key
    }
    
    return 0
}

File: Array/011-SingleNumber/SingleNumber.js
var singleNumber = function(nums) {
    let numMap = new Map()
    
    for (let i=0; i<nums.length; i++) {
        if (numMap.has(nums[i])) {
            numMap.delete(nums[i])
        } else {
            numMap.set(nums[i], 1)
        }
    }
    
    const keys = [...numMap.keys()]
    return keys[0]
};
